### ISSUE TITLE
[Feature request] Promote `ignore_index` in NegativeLogLikelihoodLoss-13 to input

### ISSUE THREAD

**Issue by justinchuby** [onnx/onnx#4993]
### What is the problem that this feature solves?

In PyTorch, `ignore_index` is dynamic: `nll_loss(Tensor self, Tensor target, Tensor? weight=None, int reduction=Mean, SymInt ignore_index=-100) -> Tensor`. To fully support this op, we can promote `ignore_index` in NegativeLogLikelihoodLoss-13 to input.

cc @gramalingam 

### Are you willing to contribute it (Y/N)

None

**Comment by titaiwangms** [comment]
cc @gramalingam

**Comment by gramalingam** [comment]
Yes, it sounds reasonable to do this. I believe onnxruntime uses custom ops for this reason (see [here](https://github.com/microsoft/onnxruntime/blob/main/orttraining/orttraining/core/graph/training_op_defs.cc#L4125)), We should consider using that extension as a candidate and propose a PR.
